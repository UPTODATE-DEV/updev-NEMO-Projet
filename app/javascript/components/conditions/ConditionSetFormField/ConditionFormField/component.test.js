import React from 'react';
import { shallow } from 'enzyme';

import ConditionSetModel from '../model';
import Component from './component';

let defaultProps;

beforeEach(() => {
  defaultProps = {
    conditionSetStore: new ConditionSetModel({
      namePrefix: 'questioning[display_conditions_attributes]',
      refableQings: [
        {
          id: '1',
          code: 'One',
          fullDottedRank: '1',
          rank: 1,
          qtypeName: 'integer',
          textual: false,
          numeric: true,
        }, {
          id: '2',
          code: 'Two',
          fullDottedRank: '2',
          rank: 2,
          qtypeName: 'text',
          textual: true,
          numeric: false,
        },
      ],
    }),
    condition: {
      leftQingId: '1',
      op: 'eq',
      operatorOptions: [
        { name: '= equals', id: 'eq' },
        { name: 'â‰  does not equal', id: 'neq' },
      ],
      refableQings: [],
      rightQingOptions: [],
    },
    index: 5,
  };
});

describe('with rightSideType as literal', () => {
  beforeEach(() => {
    defaultProps.condition.rightSideType = 'literal';
  });

  describe('with non-select leftQing', () => {
    beforeEach(() => {
      defaultProps.condition.value = '123';
    });

    it('sends correct props to ConditionValueField', () => {
      const wrapper = shallow(<Component {...defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with select leftQing', () => {
    beforeEach(() => {
      defaultProps.condition.optionSetId = '1a2';
      defaultProps.condition.optionNodeId = '7b5';
    });

    it('sends correct props to ConditionValueField', () => {
      const wrapper = shallow(<Component {...defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('with rightSideType as qing', () => {
  beforeEach(() => {
    defaultProps.condition.rightSideType = 'qing';
  });

  describe('with no applicable rightQingOptions', () => {
    beforeEach(() => {
      defaultProps.condition.rightQingId = '2';
    });

    it('renders no rightQingType dropdown', () => {
      const wrapper = shallow(<Component {...defaultProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('with applicable rightQingOptions', () => {
    beforeEach(() => {
      defaultProps.condition.rightQingOptions = [{
        id: '3',
        code: 'Three',
        fullDottedRank: '3',
        rank: 3,
        qtypeName: 'integer',
        textual: false,
        numeric: true,
      }];
      defaultProps.condition.leftQingId = '3';
    });

    describe('with forceRightSideLiteral true', () => {
      beforeEach(() => {
        defaultProps.conditionSetStore.forceRightSideLiteral = true;
      });

      it('renders no rightQingType dropdown', () => {
        const wrapper = shallow(<Component {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('with forceRightSideLiteral false', () => {
      it('renders rightQingType dropdown and rightQing dropdown', () => {
        const wrapper = shallow(<Component {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
