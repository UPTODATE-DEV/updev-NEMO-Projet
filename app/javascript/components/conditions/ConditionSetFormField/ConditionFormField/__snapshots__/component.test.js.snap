// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with rightSideType as literal with non-select leftQing sends correct props to ConditionValueField 1`] = `
<div
  className="condition-fields"
  style={
    Object {
      "display": "",
    }
  }
>
  <input
    id="questioning_display_conditions_attributes__5__id"
    key="questioning_display_conditions_attributes__5__id"
    name="questioning[display_conditions_attributes][5][id]"
    type="hidden"
    value=""
  />
  <input
    id="questioning_display_conditions_attributes__5___destroy"
    key="questioning_display_conditions_attributes__5___destroy"
    name="questioning[display_conditions_attributes][5][_destroy]"
    type="hidden"
    value="0"
  />
  <FormSelect
    includeBlank={true}
    key="questioning_display_conditions_attributes__5__left_qing_id"
    name="questioning[display_conditions_attributes][5][left_qing_id]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "1",
          "key": "1",
          "name": "1. One",
        },
        Object {
          "id": "2",
          "key": "2",
          "name": "2. Two",
        },
      ]
    }
    prompt="Choose question ..."
    value="1"
  />
  <FormSelect
    includeBlank={false}
    key="questioning_display_conditions_attributes__5__op"
    name="questioning[display_conditions_attributes][5][op]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "eq",
          "name": "= equals",
        },
        Object {
          "id": "neq",
          "name": "≠ does not equal",
        },
      ]
    }
    value="eq"
  />
  <div
    className="condition-right-side"
  >
    <div
      className="condition-value"
    >
      <ConditionValueField
        for="questioning_display_conditions_attributes__5__value"
        id="questioning_display_conditions_attributes__5__value"
        key="questioning_display_conditions_attributes__5__value"
        name="questioning[display_conditions_attributes][5][value]"
        onChange={[Function]}
        type="text"
        value="123"
      />
    </div>
  </div>
  <div
    className="condition-remove"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </a>
  </div>
</div>
`;

exports[`with rightSideType as literal with select leftQing sends correct props to ConditionValueField 1`] = `
<div
  className="condition-fields"
  style={
    Object {
      "display": "",
    }
  }
>
  <input
    id="questioning_display_conditions_attributes__5__id"
    key="questioning_display_conditions_attributes__5__id"
    name="questioning[display_conditions_attributes][5][id]"
    type="hidden"
    value=""
  />
  <input
    id="questioning_display_conditions_attributes__5___destroy"
    key="questioning_display_conditions_attributes__5___destroy"
    name="questioning[display_conditions_attributes][5][_destroy]"
    type="hidden"
    value="0"
  />
  <FormSelect
    includeBlank={true}
    key="questioning_display_conditions_attributes__5__left_qing_id"
    name="questioning[display_conditions_attributes][5][left_qing_id]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "1",
          "key": "1",
          "name": "1. One",
        },
        Object {
          "id": "2",
          "key": "2",
          "name": "2. Two",
        },
      ]
    }
    prompt="Choose question ..."
    value="1"
  />
  <FormSelect
    includeBlank={false}
    key="questioning_display_conditions_attributes__5__op"
    name="questioning[display_conditions_attributes][5][op]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "eq",
          "name": "= equals",
        },
        Object {
          "id": "neq",
          "name": "≠ does not equal",
        },
      ]
    }
    value="eq"
  />
  <div
    className="condition-right-side"
  >
    <div
      className="condition-value"
    >
      <ConditionValueField
        for="questioning_display_conditions_attributes__5__value"
        id="questioning_display_conditions_attributes__5__option_node_ids_"
        key="questioning_display_conditions_attributes__5__option_node_ids_"
        namePrefix="questioning[display_conditions_attributes][5]"
        onChange={[Function]}
        optionNodeId="7b5"
        optionSetId="1a2"
        type="cascading_select"
      />
    </div>
  </div>
  <div
    className="condition-remove"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </a>
  </div>
</div>
`;

exports[`with rightSideType as qing with applicable rightQingOptions with forceRightSideLiteral false renders rightQingType dropdown and rightQing dropdown 1`] = `
<div
  className="condition-fields"
  style={
    Object {
      "display": "",
    }
  }
>
  <input
    id="questioning_display_conditions_attributes__5__id"
    key="questioning_display_conditions_attributes__5__id"
    name="questioning[display_conditions_attributes][5][id]"
    type="hidden"
    value=""
  />
  <input
    id="questioning_display_conditions_attributes__5___destroy"
    key="questioning_display_conditions_attributes__5___destroy"
    name="questioning[display_conditions_attributes][5][_destroy]"
    type="hidden"
    value="0"
  />
  <FormSelect
    includeBlank={true}
    key="questioning_display_conditions_attributes__5__left_qing_id"
    name="questioning[display_conditions_attributes][5][left_qing_id]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "1",
          "key": "1",
          "name": "1. One",
        },
        Object {
          "id": "2",
          "key": "2",
          "name": "2. Two",
        },
      ]
    }
    prompt="Choose question ..."
    value="3"
  />
  <FormSelect
    includeBlank={false}
    key="questioning_display_conditions_attributes__5__op"
    name="questioning[display_conditions_attributes][5][op]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "eq",
          "name": "= equals",
        },
        Object {
          "id": "neq",
          "name": "≠ does not equal",
        },
      ]
    }
    value="eq"
  />
  <div
    className="condition-right-side"
  >
    <div
      className="condition-right-side-type"
    >
      <FormSelect
        includeBlank={false}
        key="questioning_display_conditions_attributes__5__right_side_type"
        name="questioning[display_conditions_attributes][5][right_side_type]"
        onChange={[Function]}
        options={
          Array [
            Object {
              "id": "literal",
              "name": "A specific value ...",
            },
            Object {
              "id": "qing",
              "name": "Another question ...",
            },
          ]
        }
        value="qing"
      />
    </div>
    <div
      className="condition-right-qing"
    >
      <FormSelect
        includeBlank={false}
        key="questioning_display_conditions_attributes__5__right_qing_id"
        name="questioning[display_conditions_attributes][5][right_qing_id]"
        onChange={[Function]}
        options={
          Array [
            Object {
              "id": "3",
              "key": "3",
              "name": "3. Three",
            },
          ]
        }
        value=""
      />
    </div>
  </div>
  <div
    className="condition-remove"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </a>
  </div>
</div>
`;

exports[`with rightSideType as qing with applicable rightQingOptions with forceRightSideLiteral true renders no rightQingType dropdown 1`] = `
<div
  className="condition-fields"
  style={
    Object {
      "display": "",
    }
  }
>
  <input
    id="questioning_display_conditions_attributes__5__id"
    key="questioning_display_conditions_attributes__5__id"
    name="questioning[display_conditions_attributes][5][id]"
    type="hidden"
    value=""
  />
  <input
    id="questioning_display_conditions_attributes__5___destroy"
    key="questioning_display_conditions_attributes__5___destroy"
    name="questioning[display_conditions_attributes][5][_destroy]"
    type="hidden"
    value="0"
  />
  <FormSelect
    includeBlank={true}
    key="questioning_display_conditions_attributes__5__left_qing_id"
    name="questioning[display_conditions_attributes][5][left_qing_id]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "1",
          "key": "1",
          "name": "1. One",
        },
        Object {
          "id": "2",
          "key": "2",
          "name": "2. Two",
        },
      ]
    }
    prompt="Choose question ..."
    value="3"
  />
  <FormSelect
    includeBlank={false}
    key="questioning_display_conditions_attributes__5__op"
    name="questioning[display_conditions_attributes][5][op]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "eq",
          "name": "= equals",
        },
        Object {
          "id": "neq",
          "name": "≠ does not equal",
        },
      ]
    }
    value="eq"
  />
  <div
    className="condition-right-side"
  >
    <div
      className="condition-value"
    >
      <ConditionValueField
        for="questioning_display_conditions_attributes__5__value"
        id="questioning_display_conditions_attributes__5__value"
        key="questioning_display_conditions_attributes__5__value"
        name="questioning[display_conditions_attributes][5][value]"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="condition-remove"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </a>
  </div>
</div>
`;

exports[`with rightSideType as qing with no applicable rightQingOptions renders no rightQingType dropdown 1`] = `
<div
  className="condition-fields"
  style={
    Object {
      "display": "",
    }
  }
>
  <input
    id="questioning_display_conditions_attributes__5__id"
    key="questioning_display_conditions_attributes__5__id"
    name="questioning[display_conditions_attributes][5][id]"
    type="hidden"
    value=""
  />
  <input
    id="questioning_display_conditions_attributes__5___destroy"
    key="questioning_display_conditions_attributes__5___destroy"
    name="questioning[display_conditions_attributes][5][_destroy]"
    type="hidden"
    value="0"
  />
  <FormSelect
    includeBlank={true}
    key="questioning_display_conditions_attributes__5__left_qing_id"
    name="questioning[display_conditions_attributes][5][left_qing_id]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "1",
          "key": "1",
          "name": "1. One",
        },
        Object {
          "id": "2",
          "key": "2",
          "name": "2. Two",
        },
      ]
    }
    prompt="Choose question ..."
    value="1"
  />
  <FormSelect
    includeBlank={false}
    key="questioning_display_conditions_attributes__5__op"
    name="questioning[display_conditions_attributes][5][op]"
    onChange={[Function]}
    options={
      Array [
        Object {
          "id": "eq",
          "name": "= equals",
        },
        Object {
          "id": "neq",
          "name": "≠ does not equal",
        },
      ]
    }
    value="eq"
  />
  <div
    className="condition-right-side"
  >
    <div
      className="condition-value"
    >
      <ConditionValueField
        for="questioning_display_conditions_attributes__5__value"
        id="questioning_display_conditions_attributes__5__value"
        key="questioning_display_conditions_attributes__5__value"
        name="questioning[display_conditions_attributes][5][value]"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="condition-remove"
  >
    <a
      onClick={[Function]}
    >
      <i
        className="fa fa-close"
      />
    </a>
  </div>
</div>
`;
